From 1e058dec3b0ff72e1edb06cac95026264f7c9bf8 Mon Sep 17 00:00:00 2001
From: James Lu <james@overdrivenetworks.com>
Date: Sun, 24 Dec 2017 11:32:37 -0800
Subject: [PATCH 4/4] Set the window icon in main, preferences, and about
 dialogs

The default window icon (at least on Xfce) isn't very informative with a busy applications panel.
---
 data/ui/kazam.ui               |  42 +++------------
 data/ui/preferences.ui         | 120 +++++++++++++++--------------------------
 kazam/frontend/about_dialog.py |   3 +-
 3 files changed, 52 insertions(+), 113 deletions(-)

diff --git a/data/ui/kazam.ui b/data/ui/kazam.ui
index 0a5345d..586f62b 100644
--- a/data/ui/kazam.ui
+++ b/data/ui/kazam.ui
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- Generated with glade 3.20.2 -->
 <interface>
-  <!-- interface-requires gtk+ 3.0 -->
+  <requires lib="gtk+" version="3.0"/>
   <object class="GtkAdjustment" id="adjustment_delay">
     <property name="upper">10</property>
     <property name="value">5</property>
@@ -13,6 +14,7 @@
     <property name="can_focus">False</property>
     <property name="title" translatable="yes">Kazam</property>
     <property name="resizable">False</property>
+    <property name="icon_name">kazam</property>
     <signal name="delete-event" handler="cb_delete_event" swapped="no"/>
     <child>
       <object class="GtkGrid" id="MainGrid">
@@ -52,8 +54,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">9</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -69,8 +69,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">8</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -86,15 +84,12 @@
                 <property name="margin_right">10</property>
                 <property name="max_length">2</property>
                 <property name="invisible_char">•</property>
-                <property name="invisible_char_set">True</property>
                 <property name="adjustment">adjustment_delay</property>
                 <signal name="value-changed" handler="cb_spinbutton_delay_change" swapped="no"/>
               </object>
               <packing>
                 <property name="left_attach">1</property>
                 <property name="top_attach">0</property>
-                <property name="width">1</property>
-                <property name="height">1</property>
               </packing>
             </child>
             <child>
@@ -102,22 +97,18 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="margin_right">5</property>
-                <property name="xalign">0.47999998927116394</property>
                 <property name="label" translatable="yes">Seconds to wait before capturing:</property>
+                <property name="xalign">0.47999998927116394</property>
               </object>
               <packing>
                 <property name="left_attach">0</property>
                 <property name="top_attach">0</property>
-                <property name="width">1</property>
-                <property name="height">1</property>
               </packing>
             </child>
           </object>
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">7</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -148,8 +139,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">3</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -165,8 +154,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">2</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -182,8 +169,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -218,8 +203,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -235,8 +218,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -248,8 +229,8 @@
               <object class="GtkLabel" id="label5">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="xalign">0.47999998927116394</property>
                 <property name="label" translatable="yes">Screenshot</property>
+                <property name="xalign">0.47999998927116394</property>
               </object>
               <packing>
                 <property name="position">1</property>
@@ -266,8 +247,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">6</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -281,8 +260,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">5</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -294,8 +271,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">3</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -308,8 +283,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">2</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -323,8 +296,6 @@
           <packing>
             <property name="left_attach">0</property>
             <property name="top_attach">4</property>
-            <property name="width">1</property>
-            <property name="height">1</property>
           </packing>
         </child>
         <child>
@@ -335,5 +306,8 @@
         </child>
       </object>
     </child>
+    <child type="titlebar">
+      <placeholder/>
+    </child>
   </object>
 </interface>
diff --git a/data/ui/preferences.ui b/data/ui/preferences.ui
index e394b56..ce27370 100644
--- a/data/ui/preferences.ui
+++ b/data/ui/preferences.ui
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- Generated with glade 3.20.2 -->
 <interface>
-  <!-- interface-requires gtk+ 3.0 -->
+  <requires lib="gtk+" version="3.0"/>
   <object class="GtkAdjustment" id="adjustment_framerate">
     <property name="lower">1</property>
     <property name="upper">60</property>
@@ -25,6 +26,7 @@
     <property name="title" translatable="yes">Kazam Preferences</property>
     <property name="resizable">False</property>
     <property name="window_position">mouse</property>
+    <property name="icon_name">kazam</property>
     <property name="gravity">south-east</property>
     <signal name="delete-event" handler="cb_delete_event" swapped="no"/>
     <child>
@@ -60,15 +62,13 @@
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Speakers:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -82,8 +82,6 @@
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -96,8 +94,6 @@
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -110,21 +106,19 @@
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
                   <object class="GtkVolumeButton" id="volumebutton_audio">
                     <property name="visible">True</property>
                     <property name="can_focus">True</property>
+                    <property name="focus_on_click">False</property>
                     <property name="receives_default">True</property>
                     <property name="has_tooltip">True</property>
                     <property name="tooltip_markup" translatable="yes">Set volume</property>
                     <property name="tooltip_text" translatable="yes">Set volume</property>
                     <property name="halign">end</property>
                     <property name="relief">none</property>
-                    <property name="focus_on_click">False</property>
                     <property name="orientation">vertical</property>
                     <property name="size">large-toolbar</property>
                     <property name="adjustment">adjustment_volume</property>
@@ -132,27 +126,36 @@
 audio-volume-high-symbolic
 audio-volume-low-symbolic
 audio-volume-medium-symbolic</property>
-                    <property name="use_symbolic">True</property>
                     <signal name="value-changed" handler="cb_volume_changed" swapped="no"/>
+                    <child internal-child="plus_button">
+                      <object class="GtkButton">
+                        <property name="can_focus">False</property>
+                        <property name="receives_default">False</property>
+                      </object>
+                    </child>
+                    <child internal-child="minus_button">
+                      <object class="GtkButton">
+                        <property name="can_focus">False</property>
+                        <property name="receives_default">False</property>
+                      </object>
+                    </child>
                   </object>
                   <packing>
                     <property name="left_attach">2</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
                   <object class="GtkVolumeButton" id="volumebutton_audio2">
                     <property name="visible">True</property>
                     <property name="can_focus">True</property>
+                    <property name="focus_on_click">False</property>
                     <property name="receives_default">True</property>
                     <property name="has_tooltip">True</property>
                     <property name="tooltip_markup" translatable="yes">Set volume</property>
                     <property name="tooltip_text" translatable="yes">Set volume</property>
                     <property name="halign">end</property>
                     <property name="relief">none</property>
-                    <property name="focus_on_click">False</property>
                     <property name="orientation">vertical</property>
                     <property name="size">large-toolbar</property>
                     <property name="adjustment">adjustment_volume2</property>
@@ -160,14 +163,23 @@ audio-volume-medium-symbolic</property>
 audio-volume-high-symbolic
 audio-volume-low-symbolic
 audio-volume-medium-symbolic</property>
-                    <property name="use_symbolic">True</property>
                     <signal name="value-changed" handler="cb_volume2_changed" swapped="no"/>
+                    <child internal-child="plus_button">
+                      <object class="GtkButton">
+                        <property name="can_focus">False</property>
+                        <property name="receives_default">False</property>
+                      </object>
+                    </child>
+                    <child internal-child="minus_button">
+                      <object class="GtkButton">
+                        <property name="can_focus">False</property>
+                        <property name="receives_default">False</property>
+                      </object>
+                    </child>
                   </object>
                   <packing>
                     <property name="left_attach">2</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -207,15 +219,13 @@ audio-volume-medium-symbolic</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
                     <property name="valign">center</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Countdown splash:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -232,8 +242,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -280,15 +288,13 @@ audio-volume-medium-symbolic</property>
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Framerate:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -301,29 +307,24 @@ audio-volume-medium-symbolic</property>
                     <property name="halign">start</property>
                     <property name="max_length">2</property>
                     <property name="invisible_char">•</property>
-                    <property name="invisible_char_set">True</property>
                     <property name="adjustment">adjustment_framerate</property>
                     <signal name="value-changed" handler="cb_spinbutton_framerate_change" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
                   <object class="GtkLabel" id="label9">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Record with:</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -335,8 +336,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -375,15 +374,13 @@ audio-volume-medium-symbolic</property>
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Automatic file saving:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -396,8 +393,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -411,8 +406,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -421,19 +414,16 @@ audio-volume-medium-symbolic</property>
                     <property name="can_focus">False</property>
                     <property name="halign">start</property>
                     <property name="valign">start</property>
-                    <property name="orientation">vertical</property>
                     <property name="action">select-folder</property>
                     <property name="local_only">False</property>
                     <property name="title" translatable="yes">Select a directory</property>
                     <property name="width_chars">20</property>
-                    <signal name="file-set" handler="cb_filechooser_video" swapped="no"/>
                     <signal name="current-folder-changed" handler="cb_filechooser_video" swapped="no"/>
+                    <signal name="file-set" handler="cb_filechooser_video" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -447,8 +437,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">2</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -459,16 +447,13 @@ audio-volume-medium-symbolic</property>
                     <property name="valign">start</property>
                     <property name="max_length">40</property>
                     <property name="invisible_char">•</property>
-                    <property name="invisible_char_set">True</property>
                     <property name="caps_lock_warning">False</property>
-                    <signal name="changed" handler="cb_entry_autosave_video" swapped="no"/>
                     <signal name="activate" handler="cb_entry_autosave_video" swapped="no"/>
+                    <signal name="changed" handler="cb_entry_autosave_video" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">2</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -521,15 +506,13 @@ audio-volume-medium-symbolic</property>
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Shutter sound:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -542,8 +525,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -557,23 +538,17 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
                   <object class="GtkComboBoxText" id="combobox_shutter_type">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
-                    <property name="entry_text_column">0</property>
-                    <property name="id_column">1</property>
                     <signal name="changed" handler="cb_shutter_type" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -612,15 +587,13 @@ audio-volume-medium-symbolic</property>
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
                     <property name="halign">end</property>
-                    <property name="xalign">1</property>
                     <property name="label" translatable="yes">Automatic file saving:</property>
                     <property name="justify">right</property>
+                    <property name="xalign">1</property>
                   </object>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -633,8 +606,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">0</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -648,8 +619,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -658,19 +627,16 @@ audio-volume-medium-symbolic</property>
                     <property name="can_focus">False</property>
                     <property name="halign">start</property>
                     <property name="valign">start</property>
-                    <property name="orientation">vertical</property>
                     <property name="action">select-folder</property>
                     <property name="local_only">False</property>
                     <property name="title" translatable="yes">Select a directory</property>
                     <property name="width_chars">20</property>
-                    <signal name="file-set" handler="cb_filechooser_picture" swapped="no"/>
                     <signal name="current-folder-changed" handler="cb_filechooser_picture" swapped="no"/>
+                    <signal name="file-set" handler="cb_filechooser_picture" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">1</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -684,8 +650,6 @@ audio-volume-medium-symbolic</property>
                   <packing>
                     <property name="left_attach">0</property>
                     <property name="top_attach">2</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
                 <child>
@@ -696,15 +660,12 @@ audio-volume-medium-symbolic</property>
                     <property name="valign">start</property>
                     <property name="max_length">40</property>
                     <property name="invisible_char">•</property>
-                    <property name="invisible_char_set">True</property>
                     <property name="caps_lock_warning">False</property>
                     <signal name="activate" handler="cb_entry_autosave_picture" swapped="no"/>
                   </object>
                   <packing>
                     <property name="left_attach">1</property>
                     <property name="top_attach">2</property>
-                    <property name="width">1</property>
-                    <property name="height">1</property>
                   </packing>
                 </child>
               </object>
@@ -734,5 +695,8 @@ audio-volume-medium-symbolic</property>
         </child>
       </object>
     </child>
+    <child type="titlebar">
+      <placeholder/>
+    </child>
   </object>
 </interface>
diff --git a/kazam/frontend/about_dialog.py b/kazam/frontend/about_dialog.py
index 6c684ed..309f29c 100644
--- a/kazam/frontend/about_dialog.py
+++ b/kazam/frontend/about_dialog.py
@@ -71,7 +71,8 @@ def AboutDialog(icons):
         pass
     # dialog.set_modal(True)
     # dialog.set_transient_for(kazam)
-    # dialog.set_decorated(False) 
+    # dialog.set_decorated(False)
+    dialog.set_icon_name("kazam")  # Sets the icon for the window
     dialog.show_all()
     dialog.set_position(Gtk.WindowPosition.CENTER)
     dialog.run()
-- 
2.15.1

